src/figuresClasses.ts

```diff
@@ -1,11 +1,101 @@
-export interface Figure {}
+export interface Figure {
+  shape: `triangle` | `circle` | `rectangle`;
+  color: `red` | `green` | `blue`;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a === 0 || b === 0 || c === 0) {
+      throw new Error('Sides can not be equal to zero');
+    }
+
+    if (a + b <= c || c + b <= a || a + c <= b) {
+      throw new Error(
+        'the longest side of a triangle is >= than a sum of two others',
+      );
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
[CODE KNOWLEDGE] - In TypeScript, when you declare a constructor parameter with an access modifier (e.g., public, private), TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You can simplify your constructor by leveraging this feature and removing the manual assignments in lines 34-38.